pekko-http-core/src/main/scala/akka/http/impl/model/parser/HeaderParser.scala

```diff
@@ -108,25 +108,25 @@ private[http] object HeaderParser {
 
   object EmptyCookieException extends SingletonException("Cookie header contained no parsable cookie values.")
 
-  def lookupParser(headerName: String, settings: Settings = DefaultSettings): Option[String => HeaderParser#Result] =
-    dispatch.lookup(headerName).map { runner => (value: String) =>
```

Comment:
We should keep this optimization, maybe we have to keep at our own version of `DynamicRuleDispatch`?

Reply:
Maybe, let's try to upstream the optimization and go back to the simpler version for now (which adds an extra hashmap lookup per header parsed).

Reply:
Makes sense to me

Reply:
> Maybe, let's try to upstream the optimization and go back to the simpler version for now (which adds an extra hashmap lookup per header parsed).

If we do this then lets make an issue on it.

Reply:
Created https://github.com/apache/incubator-pekko-http/issues/45